numpy/ma/core.py

```diff
@@ -1682,10 +1689,29 @@ def make_mask_none(newshape, dtype=None):
 
     """
     if dtype is None:
-        result = np.zeros(newshape, dtype=MaskType)
+        dtype = MaskType
     else:
-        result = np.zeros(newshape, dtype=make_mask_descr(dtype))
-    return result
+        dtype = make_mask_descr(dtype)
+
+    if not readonly:
+        return np.zeros(newshape, dtype=dtype)
+
+    else:
+        # build a buffer of the single scalar mask
+        if dtype == MaskType:
+            buff = np.ma.nomask
+        else:
+            buff = np.zeros((), dtype=dtype)
+            buff.flags.writeable = False
+
+        # duplicate it using zero strides
+        # like np.lib.stride_tricks.as_strided, but faster
+        return np.ndarray(
```

Comment:
This is faster than `np.lib.stride_tricks.as_strided` and `np.broadcast_to`, surprisingly.

It's still pretty darn slow though, when all we really want to do is directly write to the `shape` and `stride` attributes.

Could `np.zeros` acquire a `readonly` kwarg, to do this C-side?